File: prime/primes.js
function calculatePrimes() {
    const start = Date.now();
    let count = 0;
    for (let i = 1; i <= 10_000_000; i++) {
        if (isPrime(i)) {
            count++;
        }
    }
    console.log(`JavaScript:\tThere are ${count} prime numbers between 1 and 10_000_000. It took ${(Date.now() - start) / 1000} s.`);
}

function isPrime(n) {
    if (n <= 1) {
        return false;
    }
    for (let i = 2; i < Math.sqrt(n) + 1; i++) {
        if (n % i === 0) {
            return false;
        }
    }
    return true;
}

calculatePrimes();
calculatePrimes();
calculatePrimes();
calculatePrimes();
calculatePrimes();

File: prime/primes.rs
use std::time::Instant;

fn main() {
    calculate_primes();
    calculate_primes();
    calculate_primes();
    calculate_primes();
    calculate_primes();
}

fn calculate_primes() {
    let start = Instant::now();
    let mut count = 0;
    for i in 1..10_000_000 {
        if is_prime(i) {
            count += 1;
        }
    }
    println!("Rust:      \tThere are {} prime numbers between 1 and 10_000_000. It took {:.3} s.", count, start.elapsed().as_secs_f64());
}

fn is_prime(n: i32) -> bool {
    if n <= 1 {
        return false;
    }
    for i in 2..((n as f64).sqrt() as i32 + 1) {
        if n % i == 0 {
            return false;
        }
    }
    true
}
